crates/cxx-qt-lib/src/core/qurl.cpp

```diff
@@ -34,17 +34,187 @@ qurlInitFromString(::rust::Str string)
   return QUrl(qstringInitFromRustString(string));
 }
 
+::rust::String
+qurlToRustString(const QUrl& url)
+{
+  // Note that this changes UTF-16 to UTF-8
+  return qstringToRustString(url.toString());
+}
+
+QString
+qurlAuthority(const QUrl& url)
+{
+  return url.authority();
+}
+
+QString
+qurlFileName(const QUrl& url)
+{
+  return url.fileName();
+}
+
+QString
+qurlFragment(const QUrl& url)
+{
+  return url.fragment();
+}
```

Comment:
It's interesting that CXX can't wrap these, as the ComponentFormattingOptions do have a default value for each function.
So I was expecting C++ to generate two versions of this function, but apparently that's not the case and the compiler inserts the appropriate argument at the callsite.
Fascinating.

Reply:
The bigger problem was that `ComponentFormattingOptions` is a QFlags and not an Enum, how do we represent flags with CXX ? Otherwise we could add `ComponentFormattingOptions`.

And yes CXX gets confused with defaults as well, another possible thing to look at. Even if you could specify the default in the Rust definition of the bridge.